web/src/main/java/org/cbioportal/web/StudyViewController.java

```diff
@@ -571,6 +574,271 @@ private static double parseValueLinear(ClinicalData c) {
+        @ApiIgnore // prevent reference to this attribute in the swagger-ui interface. this attribute is needed for the @PreAuthorize tag above.
+        @Valid @RequestAttribute(required = false, value = "interceptedStudyViewFilter") StudyViewFilter interceptedStudyViewFilter,
+        @ApiParam(required = true, value = "Study view filter")
+        @Valid @RequestBody(required = false) StudyViewFilter studyViewFilter) {
+
+        ClinicalViolinPlotData result = new ClinicalViolinPlotData();
+        result.setAxisStart(Double.POSITIVE_INFINITY);
+        result.setAxisEnd(Double.NEGATIVE_INFINITY);
+        result.setRows(new ArrayList<>());
+
+        List<String> studyIds = new ArrayList<>();
+        List<String> sampleIds = new ArrayList<>();
+        studyViewFilterUtil.extractStudyAndSampleIds(studyViewFilterApplier.apply(interceptedStudyViewFilter), studyIds, sampleIds);
+        //DensityPlotData result = new DensityPlotData();
+        //result.setBins(new ArrayList<>());
+        if (sampleIds.isEmpty()) {
+            return new ResponseEntity<>(result, HttpStatus.OK);
+        }
+
+        List<String> sampleAttributeIds = new ArrayList<>();
+        List<String> patientAttributeIds = new ArrayList<>();
+
+        List<ClinicalAttribute> clinicalAttributes = clinicalAttributeService
+            .getClinicalAttributesByStudyIdsAndAttributeIds(studyIds,
+                Arrays.asList(categoricalAttributeId, numericalAttributeId));
+
+        clinicalAttributeUtil.extractCategorizedClinicalAttributes(clinicalAttributes, sampleAttributeIds, patientAttributeIds, patientAttributeIds);
+
+        List<String> patientIds = new ArrayList<>();
+        List<String> studyIdsOfPatients = new ArrayList<>();
+        Map<String, Map<String, List<Sample>>> patientToSamples = null;
+
+
+        if (CollectionUtils.isNotEmpty(patientAttributeIds)) {
+            List<Sample> samples = sampleService.fetchSamples(studyIds, sampleIds, Projection.DETAILED.name());
+            List<Patient> patients = patientService.getPatientsOfSamples(studyIds, sampleIds);
+            patientIds = patients.stream().map(Patient::getStableId).collect(Collectors.toList());
+            studyIdsOfPatients = patients.stream().map(Patient::getCancerStudyIdentifier).collect(Collectors.toList());
+            patientToSamples = samples.stream().collect(
+                Collectors.groupingBy(Sample::getPatientStableId, Collectors.groupingBy(Sample::getCancerStudyIdentifier))
+            );
+        }
+
+        List<ClinicalData> clinicalDataList = clinicalDataFetcher.fetchClinicalData(
+            studyIds, sampleIds, patientIds, studyIdsOfPatients, sampleAttributeIds, patientAttributeIds, null
+        );
+
+        List<ClinicalData> sampleClinicalDataList;
+        // put all clinical data into sample form
+        if (CollectionUtils.isNotEmpty(patientAttributeIds)) {
```

Comment:
If this line if the same as line 631, maybe we can put the contents into the same if clause?

Reply:
I see... There are some common stuff in the first `if` at line 631, maybe that's the reason not combining them together.

Reply:
its' because I want to call the clinical data fetcher regardless